### PR TITLE
solidity guidance

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol [hunk 1]
```diff
@@ -51,9 +52,10 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
   uint256 internal s_fallbackLinkPrice; // not in config object for gas savings
   uint96 internal s_ownerLinkBalance;
   uint256 internal s_expectedLinkBalance;
-  address internal s_transcoder;
-  address internal s_registrar;
+  address internal s_transcoder; // use interface as type
+  address internal s_registrar; // cannot use interface as type bc only use case is to check if msg.sender == s_registrar
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryBase.sol:56]: nit. "intentionally not using interface as only use case is to check if msg.sender == s_registrar"

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol [hunk 2]
```diff
@@ -51,9 +52,10 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
   uint256 internal s_fallbackLinkPrice; // not in config object for gas savings
   uint96 internal s_ownerLinkBalance;
   uint256 internal s_expectedLinkBalance;
-  address internal s_transcoder;
-  address internal s_registrar;
+  address internal s_transcoder; // use interface as type
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryBase.sol:55]: is this a todo? How is this address used currently?

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol [hunk 3]
```diff
@@ -51,9 +52,10 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
...
+  address internal s_registrar; // cannot use interface as type bc only use case is to check if msg.sender == s_registrar
 
+  // all the interfaces should be named like: ILinkToken, IAggregatorV3
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryBase.sol:58]: Let's not change this now as it would be a much bigger change in multiple unrelated files

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol [hunk 4]
```diff
@@ -62,6 +64,7 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
   PaymentModel public immutable PAYMENT_MODEL;
   uint256 public immutable REGISTRY_GAS_OVERHEAD;
 
+  // sort errors?
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryBase.sol:67]: sure :)

File: contracts/src/v0.8/dev/interfaces/KeeperRegistryInterfaceDev.sol [hunk 5]
```diff
@@ -76,6 +76,7 @@ struct Upkeep {
 }
 
 interface KeeperRegistryBaseInterface {
+  // split this to a new interface for keeper registrar
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/interfaces/KeeperRegistryInterfaceDev.sol:79]: We can do this later probably since it's needed for registrar

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol (hunk 3 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/dev/KeeperRegistryBase.sol:58]: OK. this can be a TODO

File: contracts/src/v0.8/dev/interfaces/KeeperRegistryInterfaceDev.sol (hunk 5 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/dev/interfaces/KeeperRegistryInterfaceDev.sol:79]: Yes

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol (hunk 1 above)
Comment (infiloop2) [reply contracts/src/v0.8/dev/KeeperRegistryBase.sol:56]: ah realised this was more of a q. I think it's fine to not have any comment here

File: contracts/src/v0.8/dev/KeeperRegistryBase.sol (hunk 3 above)
Comment (infiloop2) [reply contracts/src/v0.8/dev/KeeperRegistryBase.sol:58]: Yeah i think there's a bigger effort already going on to rename all interfaces with I for the whole repo

Comment (alexroan) [reply contracts/src/v0.8/dev/KeeperRegistryBase.sol:58]: I second @infiloop2 's point here. Please use the existing naming format for now.
